### PR TITLE
feat: create components-props-analyzer package

### PR DESCRIPTION
**Pull request by gdostie**
### Proposed Changes

I extracted the logic of PropsTable into a separate package called `@coveord/plasma-components-props-analyzer`. That way it shifts the work of parsing the props to compile time instead of runtime.

Right now the package isn't used in the website. But when it will be used for each demo page the props table should load instantly.

Currently

https://user-images.githubusercontent.com/35579930/200076301-371b05c4-f264-4b16-a0b2-ef4b0c1f2e82.mov

Soon

https://user-images.githubusercontent.com/35579930/200075931-ca76ec37-14b5-4d81-a5a4-4900edff089d.mov

### Potential Breaking Changes

None

### Acceptance Criteria

-   [ ] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/components-props-analyzer/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+src/components
```
Comment (aiheon) [inline packages/components-props-analyzer/.gitignore:1]:
```suggestion
src/components

```

File: packages/components-props-analyzer/src/ComponentsList.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+export interface Component {
+    /**
+     * The name of the component as it is exported
+     */
+    name: string;
+    /**
+     * the name of the package from which the component is exported
```
Comment (aiheon) [inline packages/components-props-analyzer/src/ComponentsList.ts:7]:
```suggestion
     * The name of the package from which the component is exported
```

File: packages/components-props-analyzer/src/ComponentsList.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    description: string;
+    type: string;
+    optional: boolean;
+    defaultValue: string | null;
+    deprecation: string | null;
+    params: Array<{parameterName: string; text: string}>;
+}
+
+/**
+ * These are the components whose props are going to be automatically extracted into ./src/components
```
Comment (aiheon) [inline packages/components-props-analyzer/src/ComponentsList.ts:23]: Why those 3 only? 🤔

**Review by aiheon — APPROVED** [review]
This is a huge improvement, well done! 🎉

File: packages/components-props-analyzer/src/ComponentsList.ts (hunk 3 above)
Comment (gdostie) [reply packages/components-props-analyzer/src/ComponentsList.ts:23]: I used those to test out my package and my PR was already getting pretty big. Next PR should contain all of them

**Pull request merged**